src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java

```diff
@@ -175,11 +202,16 @@ private void updateFieldsWithItemElectronicAccessField(JsonObject itemData, List
     addSubFieldGroup(effectiveLocationSubFields, locationGroup, EffectiveLocationSubFields.getLocationValues());
     addSubFieldGroup(effectiveLocationSubFields, callNumberGroup, EffectiveLocationSubFields.getCallNumberValues());
     addSubFieldGroup(effectiveLocationSubFields, itemData, EffectiveLocationSubFields.getSimpleValues());
+    //Map location name, which changed paths in json, to 952$d
+    String subFieldValue = itemData.getJsonObject(LOCATION).getString(NAME);
```

Comment:
It looks not safe, NPE can happened. Possibly this can be wrapped with something like this:
Optional.ofNullable(itemData.getJsonObject(LOCATION)).orElse(new JsonObject()).getString(NAME) ?

Reply:
done.